### PR TITLE
Consolidate Tools for Version 3.0

### PR DESCRIPTION
**Pull request by caffix**
The command-line tools have all been consolidated into a single tool with subcommands. This is the design that will be used throughout versions 3.x.x

### CODE REVIEW

File: cmd/amass/track.go [hunk 1]
```diff
@@ -9,127 +9,126 @@ import (
...
 	// Command-line switches and provided parameters
-	help        = flag.Bool("h", false, "Show the program usage message")
-	list        = flag.Bool("list", false, "Print information for all available enumerations")
-	vprint      = flag.Bool("version", false, "Print the version number of this Amass binary")
-	dir         = flag.String("dir", "", "Path to the directory containing the graph database")
-	last        = flag.Int("last", 0, "The number of recent enumerations to include in the tracking")
-	startStr    = flag.String("start", "", "Exclude all enumerations before (format: "+timeFormat+")")
-	history     = flag.Bool("history", false, "Show the difference between all enumeration pairs")
-	domainspath = flag.String("df", "", "Path to a file providing root domain names")
+	list = flag.Bool("list", false, "Print information for all available enumerations")
```
[the remark this answers is not in the document]
Comment (caffix) [reply cmd/amass/track.go]: Thanks for catching this!

File: cmd/amass/main.go [hunk 2]
```diff
@@ -45,492 +33,87 @@ var (
...
-	blacklistpath = flag.String("blf", "", "Path to a file providing blacklisted subdomains")
 )
 
+func commandUsage(msg string, cmdFlagSet *flag.FlagSet, errBuf *bytes.Buffer) {
+	amass.PrintBanner()
+	g.Fprintf(color.Error, "Usage: %s %s\n\n", path.Base(os.Args[0]), msg)
+	cmdFlagSet.PrintDefaults()
+	g.Fprintln(color.Error, errBuf.String())
+	g.Fprintf(color.Error, "An example configuration file can be found here: \n%s\n\n", exampleConfigFileURL)
+}
```
[the remark this answers is not in the document]
Comment (caffix) [reply cmd/amass/main.go]: Excellent suggestion! I have added it in

File: cmd/amass/db.go [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+	dbCommand.SetOutput(dbBuf)
+
+	dbCommand.BoolVar(&help1, "h", false, "Show the program usage message")
+	dbCommand.BoolVar(&help2, "help", false, "Show the program usage message")
+	dbCommand.Var(&args.Domains, "d", "Domain names separated by commas (can be used multiple times)")
+	dbCommand.BoolVar(&args.Options.ListEnumerations, "list", false, "Show the enumerations that include identified domains")
+	dbCommand.StringVar(&args.Filepaths.ConfigFile, "config", "", "Path to the INI configuration file. Additional details below")
+	dbCommand.StringVar(&args.Filepaths.Directory, "dir", "", "Path to the directory containing the graph database")
+	dbCommand.StringVar(&args.Filepaths.Domains, "df", "", "Path to a file providing root domain names")
+	dbCommand.StringVar(&args.Filepaths.Input, "i", "", "The Amass data operations JSON file")
```
Comment (fork-while-fork) [inline cmd/amass/db.go]: It's not obvious that this flag is used to import the data from the JSON file into the DB. Maybe change the text to:
`Import an Amass data operations JSON file`

File: cmd/amass/main.go [hunk 4]
```diff
@@ -45,492 +33,93 @@ var (
...
-	resolvepath   = flag.String("rf", "", "Path to a file providing preferred DNS resolvers")
-	blacklistpath = flag.String("blf", "", "Path to a file providing blacklisted subdomains")
 )
 
+func commandUsage(msg string, cmdFlagSet *flag.FlagSet, errBuf *bytes.Buffer) {
+	amass.PrintBanner()
+	g.Fprintf(color.Error, "Usage: %s %s\n\n", path.Base(os.Args[0]), msg)
+	cmdFlagSet.PrintDefaults()
+	g.Fprintln(color.Error, errBuf.String())
+	g.Fprintf(color.Error, "An example configuration file can be found here: \n%s\n\n", exampleConfigFileURL)
```
Comment (fork-while-fork) [inline cmd/amass/main.go]: So it's unclear what each subcommand can use from the config file. What about having an example config file for each subcommand that shows all the options supported by each subcommand?

Comment (caffix) [reply cmd/amass/main.go]: I am adding the subcommand specific info into the user guide documentation, and will add a link to the user guide to the help information

File: cmd/amass/db.go (hunk 3 above)
Comment (caffix) [reply cmd/amass/db.go]: Agreed. It has been updated

**Comment by caffix** [comment]
Thank you for the helpful feedback!

**Pull request merged**